extension/storage/local/filestorage/client.go

```diff
@@ -0,0 +1,96 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filestorage
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.etcd.io/bbolt"
+)
+
+var defaultBucket = []byte(`default`)
+
+type fileStorageClient struct {
+	db *bbolt.DB
+}
+
+func newClient(file string, timeout time.Duration) (*fileStorageClient, error) {
+	options := &bbolt.Options{Timeout: timeout}
+	db, err := bbolt.Open(file, 0666, options)
```

Comment:
I don't remember the meaning of 0666 from the top of my head. Is it read/write, non-executable?

Reply:
That's correct. This is probably too permissive even. I believe 0600 is sufficient, so I'll change it to that.